src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -150,10 +150,20 @@ public void ValidateAppInsightsTelemetryConnectionString(RuntimeConfig runtimeCo
     /// <returns>true if no validation failures, else false.</returns>
     public async Task<bool> TryValidateConfig(
         string configFilePath,
-        RuntimeConfig runtimeConfig,
         ILoggerFactory loggerFactory,
         bool isValidateOnly = false)
     {
+        RuntimeConfig runtimeConfig;
+        try
+        {
+            runtimeConfig = _runtimeConfigProvider.GetConfig();
```

Comment:
`_runtimeConfigProvider.GetConfig()` parses the config and doesn't just returned an already processed runtimeconfig object?

Reply:
it does if same instance is used. First, CLI was trying to deserialize the config which creates the dataSourceName. Now, In the engine during validation, when we create a new instance for MetadataProviderFactory, it also creates a mapping between DataSourceName and metadataProvider. This dataSourceName differs from the one created in CLI. According to our code structure `MetadataProviderFactory` should be setup first and then used to get the config which keeps thing in sync.

For your question, yes it won't parse the config again, but the mapping that exists in metadataProviderFactory will differ, if we don't first setup metadataProvider and then source config.
![image](https://github.com/Azure/data-api-builder/assets/102276754/4bc5f386-bac1-47b6-ad9d-c275a331e493)